airflow/www/views.py

```diff
@@ -1419,7 +1419,11 @@ def rendered_templates(self, session):
             ti.dag_run = DagRun(dag_id=dag_id, execution_date=dttm)
         else:
             ti = dag_run.get_task_instance(task_id=task_id, map_index=map_index, session=session)
-            ti.refresh_from_task(raw_task)
+            if ti:
+                ti.refresh_from_task(raw_task)
+            else:
+                # When there is no task instance with the given map_index
+                ti = TaskInstance(raw_task, map_index=map_index)
```

Comment:
I would prefer this to just raise directly. It is very awkward to instantiate a task instance just to fail later.

Reply:
could you check it now?
Here is the result on the UI:
![image](https://github.com/apache/airflow/assets/21311487/c78d5f55-b6b7-45d4-b146-b777fd3669cb)

Reply:
Do we want to keep the _Rendered Template_ line in the empty tab? (That’s `title` I think?)  Maybe we can change it to a `Markup` object that shows some details on the error.